packages/react-components/src/components/Attachment/AttachmentCardGroup.tsx

```diff
@@ -4,48 +4,56 @@
 import { mergeStyles, Stack } from '@fluentui/react';
 import { _pxToRem } from '@internal/acs-ui-common';
 import React from 'react';
+import { _ATTACHMENT_CARD_MARGIN_IN_PX, _ATTACHMENT_CARD_WIDTH_IN_REM } from '../styles/AttachmentCard.styles';
+import { useattachmentCardGroupLayout } from '../styles/AttachmentCardGroup.styles';
 
 /**
  * @internal
  * Props for `_AttachmentCardGroup` component.
  */
-export interface _AttachmentCardGroupProps {
-  children: React.ReactNode;
-  ariaLabel?: string;
+export enum _AttachmentCardGroupLayout {
+  /**
+   * Children are rendered in a grid layout with self resizing.
+   */
+  Grid = 'grid',
+  /**
+   * Children are rendered in a flex layout with no resizing.
+   */
+  Flex = 'flex'
 }
 
 /**
- * Note: If we use stack tokens.childrenGap, when child elements are wrapped and moved to the next line,
- * an extra margin is added to the left of each line.
- * This is a workaround to avoid this issue.
+ * @internal
+ * Props for `_AttachmentCardGroup` component.
  */
-
-const attachmentCardGroupClassName = mergeStyles({
-  flexFlow: 'row wrap',
-  '& > *': {
-    margin: _pxToRem(2)
-  },
-  /**
-   * margin for children is overriden by parent stack, so adding left margin for each child
-   */
-  '& > *:not(:first-child)': {
-    marginLeft: _pxToRem(2)
-  }
-});
+export interface _AttachmentCardGroupProps {
+  children: React.ReactNode;
+  ariaLabel?: string;
+  attachmentGroupLayout?: _AttachmentCardGroupLayout;
+}
 
 /**
  * @internal
  * Used with `_AttachmentCard` component where `_AttachmentCard` components are passed as children.
  * Renders the children equally spaced in multiple rows.
  */
 export const _AttachmentCardGroup = (props: _AttachmentCardGroupProps): JSX.Element => {
-  const { children, ariaLabel } = props;
+  const { children, ariaLabel, attachmentGroupLayout } = props;
+  const attachmentCardGroupStyles = useattachmentCardGroupLayout();
   if (!children) {
     return <></>;
   }
-
   return (
-    <Stack horizontal className={attachmentCardGroupClassName} aria-label={ariaLabel}>
+    <Stack
```

Comment:
Stack is v8 component, `mergeStyles` is v8 api but `useattachmentCardGroupLayout` uses v9 styles. Please use v8 styles for v8 components

Reply:
thanks updated